### ISSUE TITLE
fix(sidenav): do not reset focus when closed if focus is already outside of sidenav

### ISSUE THREAD

**Issue by gkalpak** [angular/material2#4203]
#### Bug, feature request, or proposal:
Bug (maybe)

#### What is the current behavior?
When `MdSidenav` (in `side` mode) is closed, it tries to return the focus back to the element that was focused before the sidenav was opened ([here][1]). This might be fine most of the time, but it is too aggressive in some cases. Consider the following usecase:

#### What is the expected behavior?
When the currently focused element is outside of a closing sidenav, the sidenav should not attempt to return the focus to the element that was focused when the sidenav opened.

#### What are the steps to reproduce?
We ran into this on angular.io. Go to [aio-staging][2], click on "Docs" in the navbar (sidenav opens), then click on "Resources" (sidenav closes), see "Docs" getting the focus (default blue outline on Chrome).

[Here][3] is a simpler reproduction.

#### Which versions of Angular, Material, OS, browsers are affected?
This is happening with v2.0.0-beta.3. All Angular/OS/browsers should be affected. Didn't check with previous versions of Material. 

[1]: https://github.com/angular/material2/blob/76faae5f378f2949bbe215df38369190773eba60/src/lib/sidenav/sidenav.ts#L146
[2]: https://aio-staging.firebaseapp.com/
[3]: http://plnkr.co/edit/T5Xu8Qy7ufNH6sm9qQBb?p=preview

**Comment by donroyco** [comment]
Issue reproduced in video (GIF):
![issue-4203](https://cloud.githubusercontent.com/assets/1763537/25307829/826cff4e-27a8-11e7-84ce-84c00b90ae1b.gif)
